### ISSUE TITLE
Authentication implementation

### ISSUE THREAD

**Issue by g4s8** [artipie/artipie#146]
Create `Authentication` implementation based on `yaml` file with credentials. I think we can use this format for credentials:
```yaml
credentials:
  - john:
      pass: "sha256:xxxxxxxxxxxxxxxxxxxxxxx"
  - jane:
      pass: "plain:qwerty"
  - joe:
      login: "joe@mail.com"
      pass: "sha256:xxxxxxxxxxxxxxxxxxxxxx"
```

So the root element is `credentials` which includes the map of unique usernames,
 each username has to have password `pass` string with format `{type}:{data}` where a type can be either `plain` or `sha256` (just for now, it can be extended later), `plain` means that the password is a plain text `data` part of `pass`, `sha256` means that we store only `SHA256` has of the password. Optionally user item may contain `login` element to override the login.

**Comment by 0crat** [comment]
The job [#146](https://github.com/artipie/artipie/issues/146) assigned to @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/0981b290-2e91-4a59-ba6f-3b89381ed01d); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be no monetary reward for this job

**Comment by olenagerasimova** [comment]
@g4s8 2 questions:
1) do not you mind if I simplify settings to 
```yaml
credentials: 
  jane: 
    pass: "plain:qwerty"
  joe: 
    login: joe@mail.com
    pass: "sha256:xxxxxxxxxxxxxxxxxxxxxx"
  john: 
    pass: "sha256:xxxxxxxxxxxxxxxxxxxxxxx"
```
2) if alternative login is set, should auth by main (joe in our case) work?

**Comment by g4s8** [comment]
@olenagerasimova 
> do not you mind if I simplify settings to

Yes, it would be better

> if alternative login is set, should auth by main (joe in our case) login work?

I think no, one user should have only one user name

**Comment by olenagerasimova** [comment]
@g4s8 forgot to ask, how we are going to locate this `credentials` file? Should the path to the file be passed via env variable?

**Comment by g4s8** [comment]
@olenagerasimova I think we may use meta config for Artipie server (see `_config.yml` file), e.g.:
```yaml
meta:
  storage:
    type: fs
    path: /var/artipie/repo
  credentials:
    # let's support only `file` type for now
    type: file
    # file location, storage key, relative to `meta.storage.path`
    path: _credentials.yml
```

**Comment by 0crat** [comment]
Job was finished in 45 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0pdd** [comment]
@g4s8 2 puzzles [#179](https://github.com/artipie/artipie/issues/179), [#180](https://github.com/artipie/artipie/issues/180) are still not solved.

**Comment by 0pdd** [comment]
@g4s8 the puzzle [#180](https://github.com/artipie/artipie/issues/180) is still not solved; solved: [#179](https://github.com/artipie/artipie/issues/179).
